feelslikemonday/app/src/main/java/com/example/feelslikemonday/ui/home/HomeFragment.java

```diff
@@ -1,42 +1,256 @@
+import android.graphics.Color;
+import android.graphics.drawable.ColorDrawable;
 import android.os.Bundle;
 import android.view.LayoutInflater;
-import android.view.MenuItem;
 import android.view.View;
 import android.view.ViewGroup;
-import android.widget.TextView;
+
 
 import androidx.annotation.Nullable;
 import androidx.annotation.NonNull;
 import androidx.fragment.app.Fragment;
 import androidx.lifecycle.Observer;
 import androidx.lifecycle.ViewModelProviders;
-import androidx.navigation.fragment.NavHostFragment;
 
-import com.example.feelslikemonday.MainActivity;
+import com.baoyz.swipemenulistview.SwipeMenu;
+import com.baoyz.swipemenulistview.SwipeMenuCreator;
+import com.baoyz.swipemenulistview.SwipeMenuItem;
+import com.baoyz.swipemenulistview.SwipeMenuListView;
+import com.example.feelslikemonday.DAO.UserCallback;
+import com.example.feelslikemonday.DAO.UserDAO;
+import com.example.feelslikemonday.DAO.VoidCallback;
 import com.example.feelslikemonday.R;
+import com.example.feelslikemonday.model.EmotionBookAdapter;
+import com.example.feelslikemonday.model.MoodEvent;
+import com.example.feelslikemonday.model.MoodType;
+import com.example.feelslikemonday.model.User;
 import com.example.feelslikemonday.ui.moods.addNewMoodActivity;
 
+import java.text.SimpleDateFormat;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.Date;
+import java.util.List;
+
 public class HomeFragment extends Fragment {
 
     private HomeViewModel homeViewModel;
+    private User currentUser;
+    private SwipeMenuListView SwipeMenuListView;
+    private List<MoodEvent> myCurrentMoodList;
+    private ArrayList<MoodEvent> myEmotionList = new ArrayList<>(); // the main Ride that handles all rides
+    private EmotionBookAdapter adapter;
 
     public View onCreateView(@NonNull LayoutInflater inflater,
                              ViewGroup container, Bundle savedInstanceState) {
         homeViewModel = ViewModelProviders.of(this).get(HomeViewModel.class);
         View root = inflater.inflate(R.layout.fragment_home, container, false);
         homeViewModel.getText().observe(getViewLifecycleOwner(), new Observer<String>() {
             @Override
-            public void onChanged(@Nullable String s) {
-            }
-        });
+            public void onChanged(@Nullable String s) { }});
         return root;
     }
 
     @Override
-    public void onViewCreated(@NonNull View view, @Nullable Bundle savedInstanceState) {
+    public void onResume() {
+        super.onResume();
+
+        UserDAO userDAO = new UserDAO();
+        userDAO.get(User.myTempUserName, new UserCallback() {
+            @Override
+            public void onCallback(User user) {
+                // current user the is object for the login user
+                currentUser = user;
+                myEmotionList.clear();
+                myCurrentMoodList = currentUser.getMoodHistory();
+                for(int i = 0; i < myCurrentMoodList.size(); i++)

```

Comment:
Would it be possible just to set myCurrentMoodList to a copy of currentUser.getMoodHistory()?